crashreports/crashreporter/CrashReportApp.cpp

```diff
@@ -0,0 +1,384 @@
+#include <wx/chartype.h>
+#include <wx/artprov.h>
+#include <wx/filename.h>
+#include <wx/stdpaths.h>
+
+#include <sstream>
+
+#include "CrashReportApp.h"
+
+#include "google_breakpad/processor/basic_source_line_resolver.h"
+#include "google_breakpad/processor/minidump_processor.h"
+#include "google_breakpad/processor/process_state.h"
+#include "google_breakpad/processor/minidump.h"
+#include "processor/stackwalk_common.h"
+
+#if defined(_WIN32)
+#include <locale>
+#include <codecvt>
+#include "client/windows/sender/crash_report_sender.h"
+namespace
+{
+    std::wstring ToPlatformString(const std::string& utf8)
+    {
+        return std::wstring_convert<std::codecvt_utf8<std::wstring::traits_type::char_type>, std::wstring::traits_type::char_type>().from_bytes(utf8);
+    }
+
+    bool SendMinidump(const std::string& url, const wxString& minidumpPath, const std::map<std::string, std::string>& arguments)
+    {
+        std::map<std::wstring, std::wstring> files;
+        files[L"upload_file_minidump"] = minidumpPath.wc_str();
+
+        std::map<std::wstring, std::wstring> parameters;
+        for (auto& p : arguments)
+        {
+            parameters[ToPlatformString(p.first)] = ToPlatformString(p.second);
+        }
+
+        google_breakpad::CrashReportSender sender(L"");
+
+        auto result = sender.SendCrashReport(
+            ToPlatformString(url),
+            parameters,
+            files,
+            nullptr
+        );
+        return result == google_breakpad::RESULT_SUCCEEDED;
+    }
+}
+#else
+
+#include "common/linux/http_upload.h"
+
+namespace
+{
+    bool SendMinidump(const std::string& url, const wxString& minidumpPath, const std::map<std::string, std::string>& arguments)
+    {
+        std::map<std::string, std::string> files;
+        files["upload_file_minidump"] = minidumpPath.ToStdString();
+
+        std::string response, error;
+        bool success = google_breakpad::HTTPUpload::SendRequest(
+            url,
+            arguments,
+            files,
+            std::string(),
+            std::string(),
+            std::string(),
+            &response,
+            NULL,
+            &error);
+            
+        return success;
+    }
+}
+
+#endif
+
+IMPLEMENT_APP(CrashReportApp);
+namespace
+{
+    std::map<std::string, std::string> parseArguments(const std::string& str)
+    {
+        int TOKEN_IDENTIFIER{ 0 };
+        constexpr int TOKEN_EQ{ 1 };
+        constexpr int TOKEN_COMMA{ 2 };
+        constexpr int TOKEN_VALUE{ 3 };
+
+        int i = 0;
+
+        std::string key;
+        int state = TOKEN_COMMA;
+        std::map<std::string, std::string> result;
+        while (true)
+        {
+            if (str[i] == 0)
+                break;
+            else if (isspace(str[i]))
+                ++i;
+            else if (isalpha(str[i]))
+            {
+                if (state != TOKEN_COMMA)
+                    throw std::logic_error("malformed parameters string: unexpected identifier");
```

Comment:
Where are the catches for this and the other throw statements?

Could the crash reporter itself crash on bad inputs?  It should not.